pkg/upstream/replicated.go

```diff
@@ -71,7 +72,7 @@ type ChannelRelease struct {
 }
 
 func (this ReplicatedCursor) Equal(other ReplicatedCursor) bool {
-	return this.ChannelName == other.ChannelName && this.Cursor == other.Cursor
+	return this.ChannelID == other.ChannelID && this.ChannelName == other.ChannelName && this.Cursor == other.Cursor
```

Comment:
I think this is a candidate for breaking backwards compatibility.  Maybe `ChannelName` comparison should be used as a fallback if one of channel IDs is not available.

Reply:
i agree, i don't see this function used anywhere in the code, only in a test file, but i'll update